libs/gpu-codec/src/main/java/org/elasticsearch/gpu/codec/ES92GpuHnswVectorsWriter.java

```diff
@@ -512,6 +511,10 @@ public void mergeOneField(FieldInfo fieldInfo, MergeState mergeState) throws IOE
         // apply mergeState.docMaps internally, so vectors are returned in the final sorted document order.
         // Unlike flush(), we don't need to explicitly handle sorting here.
         try (var scorerSupplier = flatVectorWriter.mergeOneFieldToIndex(fieldInfo, mergeState)) {
+            // retrieve directly after mergeOneFieldToIndex, so we get the most recent vector values
```

Comment:
As I understood it, before every merge operation would create a new instance of `Lucene99MemorySegmentFloatVectorScorerSupplier` that will have its own `values` field. 
With this change we are reusing the same `DelegatingFlatVectorsScorer` where `values` field is shared across all merge operations,  and this is incorrect.